packages/sources/finnhub/src/transport/quote-ws.ts

```diff
@@ -39,26 +39,42 @@ export const wsTransport = new WebSocketTransport<WsEndpointTypes>({
       }
 
       if (message.type === 'trade') {
+        const results: ProviderResult<WsEndpointTypes>[] = []
+
         const trades = message.data
-        return trades.map(({ s, p, t }) => ({
-          params: splitSymbol(s),
-          response: {
-            result: p,
-            data: {
+
+        trades.forEach(({ s, p, t }) => {
+          const params = wsTransport.getReverseMapping(s)
+
+          if (!params) {
+            logger.error(`Pair not found in websocket reverse map for message symbol '${s}'`)
+            return
+          }
+
+          results.push({
+            params,
+            response: {
               result: p,
+              data: {
+                result: p,
+              },
+              timestamps: {
+                providerIndicatedTimeUnixMs: t,
+              },
             },
-            timestamps: {
-              providerIndicatedTimeUnixMs: t,
-            },
-          },
-        }))
+          })
+        })
+
+        return results
       }
 
       return
     },
   },
   builders: {
     subscribeMessage: (params) => {
+      const symbol = buildSymbol(params)
+      wsTransport.setReverseMapping(symbol, params)
```

Comment:
I don't love that we're needing to set the reverse mapping for params now, but this is needed because the `includes` run _after_ the requestTransforms, vs the `overrides` that ran before.

Reply:
Yep, this is an unfortunate side-effect of the 2-step process.